File: backend/src/main/java/de/ckollmeier/burgerexpress/backend/dto/DisplayItemInputDTO.java
package de.ckollmeier.burgerexpress.backend.dto;

import java.util.List;

public record DisplayItemInputDTO(
        String name,
        String description,
        Boolean hasActualPrice,
        String actualPrice,
        List<String> orderableItemIds,
        Boolean published,
        String categoryId
) {
}

File: frontend/src/types/DisplayItemInputDTO.ts
export type DisplayItemInputDTO = {
    name: string;
    description: string;
    categoryId: string;
    hasActualPrice: boolean;
    actualPrice: string;
    orderableItemIds: string[];
    published: boolean;
};

export type DisplayItemInputDTOWithId = DisplayItemInputDTO & {id?: string}